robocop/reports.py

```diff
@@ -228,3 +232,95 @@ def add_message(self, *args):
 
     def get_report(self) -> str:
         return f"\nReport generated by Robocop version: {__version__}"
+
+
+class TimestampReport(Report):
+    """
+    Report name: ``timestamp``
+
+    Report that returns Robocop execution timestamp.
+    Timestamp follows local time in format of 
+    `Year-Month-Day Hours(24-hour clock):Minutes:Seconds ±hh:mm UTC offset` as default.
+
+    Example::
+
+        Reported: 2022-07-10 21:25:00 +0300
+
+    Default values can be configured by 
+    ``-c/--configure`` and ``timestamp:conf:timezone="<timezone name>":format="<format string>"`` param::
+
+        robocop --configure timestamp:conf:timezone="Europe/Paris":format="%Y-%m-%d %H:%M:%S %Z %z"
+
+    This yields following timestamp report::
+         Reported: 2022-07-10 20:38:10 CEST +0200
+
+    For timezone names,
+    see: https://en.wikipedia.org/wiki/List_of_tz_database_time_zone
+
+    For timestamp formats,
+    see: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
+
+    Useful configurations::
+        Local time to ISO 8601 format:
+        robocop --configure timestamp:conf:format="%Y-%m-%dT%H:%M:%S%z"
+
+        UTC time:
+        robocop --configure timestamp:conf:timezone="UTC":format="%Y-%m-%dT%H:%M:%S %Z %z"
+
+        Timestamp with high precision:
+        robocop --configure timestamp:conf:format="%Y-%m-%dT%H:%M:%S.%f %z"
+
+        12-hour clock:
+        robocop --configure timestamp:conf:format="%Y-%m-%d %I:%M:%S %p %Z %z"
+
+        More human readable format ``On 10 July 2022 07:26:24 +0300``:
+        robocop --configure timestamp:conf:format="On %d %B %Y %H:%M:%S %z"
+    """
+
+    def __init__(self):
+        self.name = "timestamp"
+        self.description = "Returns Robocop execution timestamp."
+        self.timezone = "local"
+        self.format = "%Y-%m-%d %H:%M:%S %z"
+
+    def configure(self, name, value):
+        if name != "conf":
+            super().configure(name, value)
+        value_err = True
+        if value.startswith("timezone="):
+            conf = value.removeprefix("timezone=").split(":", maxsplit=1)
```

Comment:
Didn't know this useful utility method - though it's introduced in Python 3.9 so we can't use it (we support 3.7+).

Reply:
Ooops, I need to refactor that. Thanks for pointing out the issue.

Reply:
Actually scratch my last advice - I forgot that it's already prepared so name variable will be already either "format" or "timezone" and value will contain the value for format/timezone. It should be easier to parse it